### DISCUSSION TITLE
Dose MagicSetFormat() function of wand not support APNG format?

### DISCUSSION THREAD

**Discussion by cristalcho** [ImageMagick/ImageMagick#2815] [Help]
Hello, I know current ImageMagick convert & identify support APNG format.
But, when I try to convert image format with using wand api, it did not work.

I used the function **MagicSetFormat()** to convert gif to apng format.
But, the result is **image/x-portable-pixmap** not **image/png.**
Dose ImageMagick Wand not support APNG? How can I convert image to APNG format?

**Comment by fmw42** [comment]
You need ffmpeg installed with Imagemagick. Then preface your output with APNG:

In command line it would be:

`magick -delay 50 -loop 0 lena.jpg mandril3.jpg monet2.jpg zelda1.jpg APNG:anim.apng`

**Reply by cristalcho** [reply]
I already installed ffmpeg with Imagemagick.
When I convert using wand function, It did not work.
I tried magick convert & identify with format APNG and it worked well.

I mean, the conversion is not work that image convert to APNG with using wand api

**Comment by urban-warrior** [comment]
The APNG format must be explicit.  Instead of MagicSetFormat(), simply set your output filename with the APNG tag, e.g. APNG:anim.apng.  You also need the latest release of ImageMagick as support for APNG is a recently addition.
